assets/js/src/components/views/tab-row.js

```diff
@@ -31,20 +46,190 @@ module.exports = function( app, _meta_keys ) {
+			var component = jQuery( evt.target ).closest('.component-table-wrapper').attr('id');
+			// Update Data
+			this.model.set( 'field_subfields', {} );
+			if ( '' === value ) {
+				this.model.set( 'field_field', '' );
+				jQuery('#'+component+' .component-table-inner ').find('select').html("<option value=''>Unused</option>").val(""); 
+			} else {
+				this.model.set( 'field_field', value );
+				// Update subfields
+				this.updateAjax_ComponentSubfields(component, value, false);
+			}
+		},
+
+		changeSubfield: function( evt ) {
+			var value = jQuery( evt.target ).val();
+			var index = jQuery( evt.target ).attr('data-index');
+			var subfield_data = this.model.get( 'field_subfields');
+			if(!subfield_data){ subfield_data = {}; }
+			subfield_data[index] = value;
+			this.model.set( 'field_subfields', subfield_data );
+		},
+
+		optionBuilder: function( data ) {
+			console.log('optionBuilder');
+			var options_html = "<option value=''>Unused</option>";
+			jQuery.each(data.field_data, function(i, field) {
+				// console.log('data: ',data);
+				options_html += "<option class='hidden' value='"+field.key+"' data-type='"+field.type+"'>"+field.label+"</option>";
+			}); 
+			return options_html;
+		},
+
+		/**
+		 * AJAX Update: "Field" - ACF Field group's field
+		 * 
+		 * @param {string} component - ID without the "#" of the component parent row
+		 * @param {string} field_name - Parent field name/key of the sub fields, should be a repeater
+		 * @param {object} saved_fields - OPTIONAL: Pass saved subfields if you want to set pre-existing values
+		 */
+		updateAjax_Field: function( component, field_name, saved_fields ) {
+			saved_fields = typeof saved_fields !== 'undefined' ? saved_fields : "";
+			var $this = this;
+			// console.log('updateAjax_Field');
+			// console.log('field_name: ',field_name);
+			// console.log('saved_fields: ',saved_fields);
+
+			// Update UI
+			jQuery('#'+component+' .wp-type-field-select ~ span.select2').addClass('ajax-disabled');
+			// Get Updated Data
+			jQuery.post( window.ajaxurl, {
+				action: 'gc_component_subfields',
+				subfields_data: {
+					name: field_name,
+				}
+			}, function( response ) {
+				// Update UI
+				jQuery('#'+component+' .wp-type-field-select ~ span.select2').removeClass('ajax-disabled');
+				console.log('RESPONSE: ',response);
+
+				// SUCCESS
+				if( response.success ){
+					// Ensure response has subfield data
+					if( response.data.field_data && response.data.field_data.length ){
+						// Build options HTML:
+						var options_html = $this.optionBuilder(response.data);
+						// Inject into select fields
+						jQuery('#'+component).find('.wp-type-field-select').html(options_html);
+
+						// If existing subfields are passed, update specific dropdown options
+						if(saved_fields){
+							jQuery('#'+component).find('.wp-type-field-select').val(saved_fields);
+						}
+					}
+				}
+				// ERROR
+				else{
+					console.log('AJAX ERROR!');
```

Comment:
What do you think to outputting something to the user if it fails to ajax? Even if it is as simple as please try again or get in contact with us. Just feels like it's silently failing without any visibility.